pkgs/by-name/st/styluslabs-write/package.nix

```
@@ -0,0 +1,99 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  gitMinimal,
+  pkg-config,
+  SDL2,
+  xorg,
+  dbus,
+  libcap,
+  libGL,
+  roboto,
+  imagemagick,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "styluslabs-write";
+  version = "2024-09-26";
+
+  src = fetchFromGitHub {
+    owner = "styluslabs";
+    repo = "Write";
+    rev = "7110e5db87c154d28efd9507be4ab768af7403e0";
+    hash = "sha256-/IYI/trC14IBYtYn9LkjNrMZoCn6jtQIgq0ZqnsLISs=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./sdl2_link_dynamic.patch ];
```

Nit:

Could you add a quick explanation why the patch is necessary?

```suggestion
  patches = [
    # Take SDL from buildInputs rather than vendor source
    ./sdl2_link_dynamic.patch
  ];
```

It'd also be great if you could propose this upstream (or at least report it) and add a backlink to the comment.

---

https://github.com/styluslabs/Write/pull/3